x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -631,6 +647,56 @@ export class RulesClient {
     });
   }
 
+  public async getExecutionLogForRule({
+    id,
+    dateStart,
+    dateEnd,
+    filter,
+    page,
+    perPage,
+    sort,
+  }: GetExecutionLogByIdParams): Promise<IExecutionLogResult> {
+    this.logger.debug(`getExecutionLogForRule(): getting execution log for rule ${id}`);
+    const rule = (await this.get({ id, includeLegacyId: true })) as SanitizedRuleWithLegacyId;
+
+    // Make sure user has access to this rule
+    await this.authorization.ensureAuthorized({
+      ruleTypeId: rule.alertTypeId,
+      consumer: rule.consumer,
+      operation: ReadOperations.GetExecutionLog,
+      entity: AlertingAuthorizationEntity.Rule,
+    });
```

Comment:
I don't see any audit events for this API.

I think you need to make the following changes:

1. Wrap this AuthZ check in a try/catch, log a `failure` event here if the user isn't authorized, and re-throw the error
2. Before returning the result to the user, log a `success` event

I'm not familiar with the execution log but I guess you'll want to add a new audit event for it too. If you need to do that, remember to update the audit logging docs at `docs/user/security/audit-logging.asciidoc` too!

Reply:
Thanks @jportner! I will add a new audit event for `GET` the rule. However, after the rule GET, we are accessing the event log index, which is not a saved object (and technically in "technical preview"). Do we still need to audit access to it? Or do we only need to audit access to the saved object?